### PR TITLE
Continue upgrading build tooling

### PR DESCRIPTION
**Pull request by bdewilde**
### changes

- migrate python packaging from setup files to equivalent `pyproject.toml` file
- add github actions workflow to run tests, lint, and check types; only the tests are meant to be blocking, for now
- constrains `numpy` package dependency to avoid an error
- bumps `textacy` package dependency from 0.5 to 0.10 so we can drop a transitive dependency that refused to install
    - updates call to `textacy` functionality that changed between versions
    - replaces one call with another package's equivalent, and adds that package to the dependencies

### context

trying to move `colandr` forward in time in manageable chunks, from 2016 to 2020+. starting with `textacy` because i know it well -- i am its author :) -- with `dedupe` likely next. trying to save `flask` and its extensions until last, because that will probably be a mess.

### questions

- tests aren't running in CI because of missing envvars. i'm not sure what the best solution is: create those envars in the CI env? modify the code so it doesn't require them? something else??

### CODE REVIEW

File: colandr/api/resources/fulltext_uploads.py [hunk 1]
```diff
@@ -2,23 +2,22 @@
...
 from marshmallow import fields as ma_fields
 from marshmallow.validate import Range
 from webargs.flaskparser import use_kwargs
-
-from textacy.preprocess import fix_bad_unicode
+from werkzeug.utils import secure_filename
 
 from colandr import api_
+
 from ...lib import constants
```
Comment (samanz) [inline colandr/api/resources/fulltext_uploads.py:15]: Just a stylistic question, but should we move to Absolute imports?

**Review by samanz — APPROVED** [review]
LGTM. I think we should get away from using env variables for configs, which will also make dockerizing easier. I do wonder what the best method for config management moving forward might be.

File: colandr/api/resources/fulltext_uploads.py (hunk 1 above)
Comment (bdewilde) [reply colandr/api/resources/fulltext_uploads.py:15]: tbh i don't have a strong preference! (this is rare for me when it comes to python style, haha)

at some point i'm going to blacken this entire code base, and maybe then i'll revisit our import conventions

**Comment by bdewilde** [comment]
@samanz any thoughts on what to do about the CI testing issue? i might turn them off for now to avoid getting angry emails every commit, but would be great to get a proper CI system up and running asap.

**Pull request merged**